WordPress/gutenberg
# Popover: rewrite Storybook examples using controls

## What?


In the context of #42770, the Storybook examples are being of little help when working on the `Popover` component in isolation.

This PR refactors the Storybook examples to use controls (instead of the deprecated knobs add-on), and also rewrites some of the stories in a way that hopefully showcases the usages of the component in a better way compared to how it was before.

## Why?


Better Storybook examples = better dev experience (both as a component author, and as a component consumer).

## How?


1. Converted from knobs to controls, listing all known `Popover` props
2. The `default` example has been changed to mirror the example snippet in the component's README (see notes below). I feel like the previous default example was not really useful anyway.
3. The `positions` example has been replaced with a new story that showcases all possible values for `placement` (ie. the new prop that is supposed to replace `position`)
4. The dragging example has been removed, as I feel it didn't give much value after adding the new "default" and "all placements" story (and its code was very complex)
5. The "Dynamic height" story has been kept mostly intact
6. Added a new story to test the behavior of the popover when its anchor is inside an iframe, but the popover itself is rendered outside of that iframe

## Testing Instructions





1. Make sure that Storybook builds correctly and that all stories are accessible
2. Make sure that 

### Known issues

Working on this refactor highlighted a couple of issues:

####  The `Popover` is not anchoring itself correctly to its parent (which is the default behaviour when an explicit anchor it not passed):

https://user-images.githubusercontent.com/1083581/182440756-1d02165f-b628-45e8-9b42-7370a7e4f050.mp4

This behavior can be also spotted on `trunk` by visiting [the positioning storybook example](https://wordpress.github.io/gutenberg/?path=/story/components-popover--positioning), and noting that the `Popover` is not positioning itself correctly with respect to its parent (the pink bordered box). This is likely another regression introduced by #40740, and if folks confirm it, I will create a new issue and add it to #42770


#### The positioning of the `Popover` is a bit off in the `iframe` example

https://user-images.githubusercontent.com/1083581/182441479-25761b05-6b4d-40d6-a3c1-4b4352f74471.mp4


I haven't had time to debug this example in depth yet, but it looks like the offset applied to the `Popover` to compensate the `iFrame` positioning is not correct. It's interesting (and worrying) that the `Popover` doesn't work as expected with a simple `iframe` example. We should understand if something is wrong in the way I set up the Storybook example, or if this is another bug that we need to add to #42770 and fix.
